code-challenges.test.js

```diff
@@ -22,9 +22,47 @@ const secretCodeWord2 = "Gobbledygook"
 const secretCodeWord3 = "Eccentric"
 // Expected output: "3cc3ntr1c"
 
-
+describe("crypt", () => {
+    let secretCodeWord1 = "Lackadaisical"
+    let secretCodeWord2 = "Gobbledygook"
+    let secretCodeWord3 = "Eccentric"
+    // a test/it method, nested within the describe block, that in plain words, describes that the function does.
+    it(" takes in a string and returns a coded message. The coded message converts 'a' to 4, 'e' to 3, 'i' to 1, and 'o' to 0.", () => {
+
+        //an expect method, nested within the test block, calling on the crypt() function, followed by the .toEqual() matcher that checks the expected output of the function return.
+        expect(crypt(secretCodeWord1)).toEqual("L4ck4d41s1c4l")
+        expect(crypt(secretCodeWord2)).toEqual("G0bbl3dyg00k")
+        expect(crypt(secretCodeWord3)).toEqual("3cc3ntr1c")
+    })
+})
+// got red
 
 // b) Create the function that makes the test pass.
+//decalare a funtion, and take in the string as param
+function crypt(string){
+  // split the string to and single letters and store in the array
+      let newArr = string.split('')
+      // call for loop to search this array, if requirement meets, then take action
+      for (let i = 0; i < newArr.length; i++) {
+        if (newArr[i] === 'a' || newArr[i] === 'A'){
+           newArr[i] = '4'
+        }else if (newArr[i] === 'e' || newArr[i] === 'E'){
+           newArr[i] = '3'
+        }else if (newArr[i] === 'i' || newArr[i] === 'I'){
+           newArr[i] = '1'
+        }else if (newArr[i] === 'o' || newArr[i] === 'O'){
+           newArr[i] = '0'
+        }else{
+           // console.log('Please input a string or string variable.');
```

Comment:
It looks like your else statement is doing much. The rest of your logic looks great.